File: packages/syft/src/syft/core/node/new/serialize.py
# stdlib
from typing import Any
from typing import Type


def _serialize(
    obj: object,
    to_proto: bool = True,
    to_bytes: bool = False,
    class_type: Type = type(None),
) -> Any:
    # relative
    from .recursive import rs_object2proto

    proto = rs_object2proto(obj, class_type=class_type)

    if to_bytes:
        return proto.to_bytes()

    if to_proto:
        return proto

File: packages/syftjs/src/serde/serialize.ts
import * as capnp from 'capnp-ts';

import { RecursiveSerde } from '../capnp/recursive_serde.capnp';

import { getSerdeSchema, serializeObject } from './serializable';

/**
 * Method used to serialize a SyftJS Object or Primitives into a Cap'n Proto array buffer.
 * @param {any} obj - Object to be serialized.
 * @returns {ArrayBuffer} Array buffer with capnp structure of the serialized object.
 */
export function serialize(obj: any) {
  // Create a new Cap'n Proto message
  const message = new capnp.Message();
  const rs = message.initRoot(RecursiveSerde);

  // Get the serde schema of the object
  const serdeSchema = getSerdeSchema(obj);
  rs.setFullyQualifiedName(serdeSchema.fqn);

  // Call inner method to decide how to serialize the object (primitive/recursive).
  serializeObject(obj, serdeSchema, rs);

  return message.toArrayBuffer();
}
